src/core/init/stream_events_emitter/types.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface IStreamEventData {
+  start: number;
+  end?: number;
+  id?: string;
+  isBeingPlayed: boolean;
+  data: unknown;
+}
```

Comment:
Bear in mind that this is the object exposed in the API.

From there I have multiple problems:
  1. Why would they need an ID? The only use-case I see where it would make sense is to correlate streamEventIn and streamEventOut (note that I'm also against the streamEventOut event because it complexifies that).

  2. Why would they need a isBeingPlayed boolean?
      That sounds like future bugs (for example we could decide to deep clone our event's payload in the future, this allows to make sure an application don't break our code accidentally).

   3. I think that setting `data` to `unknown` might not be the best thing as the application might have to perform duck typing in every way possible to know the format. We actually know what format it is so we could find a way to either communicate it or convert it to a common format